### ISSUE TITLE
[Feature Request/ATB] Option to hide hide custom player refits from the purchase menu/unit cache

### ISSUE THREAD

**Issue by Tamren** [MegaMek/mekhq#356]
I like customizing units when playing ATB. Most of these are minor refits like flipping rear-firing lasers to face forwards, or adding things like CASE to canon designs. However each refit is required to have a unique name, and if you have a lot of them they tend to clutter up the purchase unit menu with custom units that should not be for sale from a factory. Custom units also show up in Megamek when you load up the cache to add units to a game.

The simplest fix I can think of is to add another flag to the Advanced Search menu that sorts out player created mech refits and other custom units. This would let you filter them out of the broader list when purchasing units or adding units to a megamek match. You could also use this in reverse to filter out everything but your custom units so that you can find them easier.

**Comment by pheonixstorm** [comment]
There is an option that should do just this. I don't recall the name but it should be in the allowed unit tab in the Campaign Options. I think it is Canon units purchase only or some such.

When using AtB the unit market shouldn't include custom units anyway, at least I have never seen any pop up before. @neoancient would know more on if that is possible via the AtB market.

**Comment by Tamren** [comment]
Sorry I mean the "purchase unit" menu, not the unit market. I wish the names were more distinct. The unit MARKET works fine, but the purchase menu gets cluttered by all of the refits.

If I set the game to use canon unit's only it does remove the custom units from the purchase unit menu. However it's not quite what I need because if I flip the canon switch it ALSO disables me from using any of my own designs in refits.  So new filters for the advanced search would still be helpful here.

**Comment by pheonixstorm** [comment]
Actually I think excluding the option from the refit menu would probably work better in this situation. I don't know what the original intention was with the canon only option but I would think it was designed to only limit the purchase unit menu and not the refit menu.

**Comment by neoancient** [comment]
So are you asking for separate options for purchases and refits, so you can exclude canon units from one or the other without it necessarily having to be both?

**Comment by Dylan-M** [comment]
@neoancient That would be "non canon" units, not "canon" ones ;)

**Comment by neoancient** [comment]
Right. That.

**Comment by Tamren** [comment]
@neoancient Yes that would work. So instead of having just the "only allow canon units for purchase" checkbox there would need to be a second one that said something like "only allow canon refits kits".
